File: app/src/main/java/com/star_zero/example/androidmvvm/domain/task/TaskId.java
package com.star_zero.example.androidmvvm.domain.task;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

import com.star_zero.example.androidmvvm.domain.Identifier;

public class TaskId extends Identifier implements Parcelable {

    public TaskId(@NonNull String value) {
        super(value);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.value);
    }

    private TaskId(Parcel in) {
        super(in.readString());
    }

    public static final Creator<TaskId> CREATOR = new Creator<TaskId>() {
        @Override
        public TaskId createFromParcel(Parcel source) {
            return new TaskId(source);
        }

        @Override
        public TaskId[] newArray(int size) {
            return new TaskId[size];
        }
    };
}

File: app/src/main/kotlin/com/star_zero/example/androidmvvm/domain/task/TaskId.kt
package com.star_zero.example.androidmvvm.domain.task

import android.os.Parcel
import android.os.Parcelable
import com.star_zero.example.androidmvvm.domain.Identifier

data class TaskId(override val value: String) : Identifier, Parcelable {
    companion object {
        @JvmField val CREATOR: Parcelable.Creator<TaskId> = object : Parcelable.Creator<TaskId> {
            override fun createFromParcel(source: Parcel): TaskId = TaskId(source)
            override fun newArray(size: Int): Array<TaskId?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(value)
    }
}